lib/interpreters/compute_several_solutions.rb

```diff
@@ -259,7 +259,10 @@ def self.select_best_heuristic(vrp)
               }
         end
       }
-      size_mtws = services.count{ |service| service.activity.timewindows.size > 1 }
+      size_mtws = services.count{ |service|
+        (service.activity&.timewindows&.size || 1) > 1 || # No timewindow means an implicit infinite timewindow
+          service.activities.flat_map{ |act| [act.timewindows.size, 1].max }.sum > 1
```

Comment:
We can simplify as follows I think. Does it look okay to you? 
```suggestion
        service.activity&.timewindows&.size.to_i > 1 || 
          # No timewindow means an implicit infinite timewindow
          service.activities.sum{ |act| [act.timewindows.size, 1].max } > 1
```